runtime/compiler/env/J9ClassEnv.cpp

```diff
@@ -464,16 +522,28 @@ J9::ClassEnv::enumerateFields(TR::Region& region, TR_OpaqueClassBlock * opaqueCl
             break;
             }
          }
-      size_t nameSize = strlen(field->name)+1;
-      char *fieldName = new (region) char[nameSize];
-      strncpy(fieldName, field->name, nameSize);
-      TR_ASSERT_FATAL(fieldName[nameSize-1] == '\0', "fieldName buffer was too small.");
-      int32_t offset = field->offset + TR::Compiler->om.objectHeaderSizeInBytes();
+
+      char *fieldName = mergeFieldNames(prefix, field->name, false /* withDotAtTheEnd */ , region);
+      int32_t offset = offsetBase + field->offset + TR::Compiler->om.objectHeaderSizeInBytes();
       bool isVolatile = (field->modifiers & J9AccVolatile) ? true : false;
       bool isPrivate = (field->modifiers & J9AccPrivate) ? true : false;
       bool isFinal = (field->modifiers & J9AccFinal) ? true : false;
+      traceMsg(comp, "type layout definingClass %p field: %s, field offset: %d offsetBase %d\n", definingClass, fieldName, field->offset, offsetBase);
```

Comment:
Likely we'd care about such information during ILGen phase, I'll guard it with `TR_TraceILGen`, although it shouldn't cause any issue since the definition of `traceMsg` checks if `(comp)->getDebug()` exists.  

https://github.com/eclipse/openj9-omr/blob/025d6b2b62be8def4c8887e5c92f71195d037042/compiler/compile/OMRCompilation.hpp#L260-L264
```c++
         #define traceMsg(comp, msg, ...) \
      ((comp)->getDebug() ? \
            (comp)->getDebug()->trace(msg, ##__VA_ARGS__) : \
            (void)0)
```

Reply:
Unguarded `traceMsg` calls don't normally cause functional issues (as you said, it checks if `comp->getDebug()` exists) but it can be a bit annoying when you get extra trace information you didn't ask for (e.g. you only specified `traceFull` but still got trace messages from ilgen). :wink: